File: epi_judge_cpp/snake_string.cc
#include <string>

#include "test_framework/generic_test.h"
using std::string;

string SnakeString(const string& s) {
  if (s.empty()) {
  return "";
  }
  string snake;
  for (auto i = 1; i < s.size(); i += 4) {
    snake.push_back(s[i]);
  }
  for (auto i = 0; i < s.size(); i += 2) {
    snake.push_back(s[i]);
  }
  for (auto i = 3; i < s.size(); i += 4) {
    snake.push_back(s[i]);
  }
  return snake;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"s"};
  return GenericTestMain(args, "snake_string.cc", "snake_string.tsv",
                         &SnakeString, DefaultComparator{}, param_names);
}

File: epi_judge_python/snake_string.py
from test_framework import generic_test


def snake_string(s: str) -> str:
    return "".join(
        [s[i] for i in range(1, len(s), 4)]
        + [s[i] for i in range(0, len(s), 2)]
        + [s[i] for i in range(3, len(s), 4)]
    )


if __name__ == "__main__":
    exit(
        generic_test.generic_test_main(
            "snake_string.py", "snake_string.tsv", snake_string
        )
    )
